### ISSUE TITLE
BLOCKINGMODE is not honored for Deep CNAME inspection.

### ISSUE THREAD

**Issue by DarkDimius** [pi-hole/pi-hole#3935]
### Versions

  Pi-hole version is v5.2.1 (Latest: v5.2.1)
  AdminLTE version is v5.2.1 (Latest: v5.2.1)
  FTL version is v5.3.2 (Latest: v5.3.2)

### Platform

Ubuntu 18.04.5 LTS

LXC container(but it shouldn't matter)

### Expected behavior

```
dig link.generalassemb.ly

; <<>> DiG 9.10.6 <<>> link.generalassemb.ly
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 28352
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1472
;; QUESTION SECTION:
link.generalassemb.ly.			2	IN	A	0.0.0.0

;; Query time: 14 msec
;; SERVER: 192.168.8.6#53(192.168.8.6)
;; WHEN: Sat Dec 19 18:39:40 PST 2020
;; MSG SIZE  rcvd: 50
```

### Actual behavior / bug

Instead of using the BLOCKINGMODE=null it uses NXDOMAIN response for queries blocked using deep domain inspection. 

```
; <<>> DiG 9.10.6 <<>> link.generalassemb.ly
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 28352
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1472
;; QUESTION SECTION:
;link.generalassemb.ly.		IN	A

;; Query time: 14 msec
;; SERVER: 192.168.8.6#53(192.168.8.6)
;; WHEN: Sat Dec 19 18:39:40 PST 2020
;; MSG SIZE  rcvd: 50
```

```
Dec 20 02:33:42 dnsmasq[334]: query[A] link.generalassemb.ly from 192.168.8.191
Dec 20 02:33:42 dnsmasq[334]: forwarded link.generalassemb.ly to 127.0.0.1
Dec 20 02:33:42 dnsmasq[334]: reply link.generalassemb.ly is <CNAME>
Dec 20 02:33:42 dnsmasq[334]: reply cb.sailthru.com is blocked during CNAME inspection
```
### Steps to reproduce

1. Make sure you have `cb.sailthru.com` blocked, either via one of lists or via dirrectly adding it to blocklist
2. do `dig link.generalassemb.ly` (which is actually a CNAME to `cb.sailthru.com`)

**Comment by DarkDimius** [comment]
Why this matters: my home network gives client a local domain prefix via http. Having `link.generalassemb.ly` fail resolution makes some clients attempt to find it at `link.generalassemb.ly.localdomain`(which resolves) and thus leads to surprises

**Comment by dschaper** [comment]
The response is `NOERROR` not `NXDOMAIN`.

**Comment by DarkDimius** [comment]
Agreed, still, the expected response under this configuration (BLOCKINGMODE=null) should be resolving to 0.0.0.0, right?

**Comment by dschaper** [comment]
Possibly, but I think the bigger problem is the fact that `link.generalassemb.ly.<local_domain>` actually resolves.

**Comment by dschaper** [comment]
What does `ping link.generalassemb.ly` show as a response?

**Comment by DarkDimius** [comment]
> Possibly, but I think the bigger problem is the fact that link.generalassemb.ly.<local_domain> actually resolves.

Yeah, I don't think I'd be able to change that. This network uses wildcard dns A record.

> What does ping link.generalassemb.ly show as a response?

```
$ ping link.generalassemb.ly
PING link.generalassemb.ly.<local_domain_redacted> (REDACTED) 56(84) bytes of data.
64 bytes from REDACTED (REDACTED): icmp_seq=1 ttl=64 time=2.78 ms
```

where REDACTED is address that wildcard A record points to.

**Comment by dschaper** [comment]
Also, can you go to your admin interface, under the Query Log, do a search for `link.generalassemb.ly` and show what the output is for status?

**Comment by dschaper** [comment]
So, anything that doesn't return an IP address is automatically returned an IP from unbound?

**Comment by DarkDimius** [comment]
> Also, can you go to your admin interface, under the Query Log, do a search for link.generalassemb.ly and show what the output is for status?

yes, more than happy to!
<img width="981" alt="image" src="https://user-images.githubusercontent.com/1084930/102704363-590f4080-422f-11eb-9a2a-37c8088cc848.png">

> So, anything that doesn't return an IP address from Pi-hole is automatically returned an IP from unbound?

yes. It's an unfortunate implication of how the network is setup.

**Comment by dschaper** [comment]
Thanks, I'll ask @DL6ER to take a look. He's extremely busy right now so it may take a bit for his review.

**Comment by DarkDimius** [comment]
<img width="987" alt="image" src="https://user-images.githubusercontent.com/1084930/102704434-0e41f880-4230-11eb-988e-f6d84b382c21.png">
is actually a better demo
